### DISCUSSION TITLE
To fork or not to fork

### DISCUSSION THREAD

**Discussion by NiceGuyIT** [Ylianst/MeshCentral#5322] [General]
Hi @Ylianst and @krayon007,

It's approaching a year since you and Bryan have changed jobs and settled into your new life. I hope things are going well and you both are enjoying life.

MeshCentral is being used by many people and the community is very grateful for what you started. Some community members are helping with triaging issues and submitting pull requests. Many thanks to @si458 for stepping up! Unfortunately, we can only do so much without access to the repository. To keep the community united, please consider adding one or two select individuals to the repository as collaborators. You can read about the [permissions collaborators have for personal repos](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/permission-levels-for-a-personal-account-repository#collaborator-access-for-a-repository-owned-by-a-personal-account). If MeshCentral was converted to an organization, organization members can have [finer grained permissions for roles](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization#permissions-for-each-role).

The alternative is to fork MeshCentral so the community can maintain it. This happened to Gogs and the community forked it to Gitea ([source](https://blog.gitea.com/welcome-to-gitea/)). Another example: KeePassX was slow to respond and the community forked it to KeePassXC ([source](https://github.com/keepassxreboot/keepassxc/issues/43)). The last example is the community fork of openhardwaremonitor to LibreHardwareMonitor ([source](https://github.com/openhardwaremonitor/openhardwaremonitor/issues/1272)). These are examples of single maintainer repositories where the maintainer didn't have the time to keep up with the pace the community desired.

As the sole maintainer of a popular repository, I ask that you let the community help _you_, help _MeshCentral_, and more importantly, help the **community**! Please consider adding someone as a contributor or convert this repo to an organization and add someone to the organization.

**Comment by si458** [comment]
~~Sadly it can't be converted to an organisation because the is already an organised and it has no repository or members which is so annoying!~~

we can't make an organisation called meshcentral because the is already one called that which is annoying but I'm sure we can look into stuff

**Reply by NiceGuyIT** [reply]
GitHub [Name Squatting Policy](https://docs.github.com/en/site-policy/other-site-policies/github-username-policy#name-squatting-policy):

> GitHub prohibits account name squatting, and account names may not be reserved or inactively held for future use. Accounts violating this name squatting policy may be removed or renamed without notice. Attempts to sell, buy, or solicit other forms of payment in exchange for account names are prohibited and may result in permanent account suspension.

Who wants to contact GitHub?

**Reply by silversword411** [reply]
Lets wait for someone official to pipe up, happy to take the bull by the horns if necessary :)

**Reply by NiceGuyIT** [reply]
[Started the ball rolling](https://github.com/orgs/community/discussions/65585). Let's see how far I get.

**Comment by silversword411** [comment]
I have reserved the https://github.com/meshcentral2 org just in case.

Happy to make ylianst or krayon007 owners

**Reply by NiceGuyIT** [reply]
How about "MeshCentral-community" or "MeshCentral-org"?

**Reply by si458** [reply]
we could do `meshcentral-community` but i would need the admin of https://meshcentral-community.com/ to email us first to make sure this was ok! dont want to step on any toes!

**Reply by silversword411** [reply]
I looked at that when I rebuilt all the docs in mkdocs on github. I wasn't going to put all that work/hours into custom wiki code that someone else was hosting, and might disappear at any time. 

Would be nice to consolidate everything under one roof...but he seemed attached to his wiki.

**Comment by tschettervictor** [comment]
Cant the repo just be transferred to a different owner? The name can be the same. Or maybe MeshCentral+

**Reply by silversword411** [reply]
Yes it can be moved from ylianst private to an org

**Comment by jammsen** [comment]
As a side-stander i would be for this, im helping admins in my organization to work with Mesh-Central (help by proxy basically) but without CVEs and @si458 was always a big help fixing thoose CVEs, while i provided the needed data. He also was so transparent and explained why @Ylianst has no time. 

You can see that and this reflects into waiting for 2 weeks+ for accepting a PR that fixes 9+ CVEs. @Ylianst isnt working for Intel anymore and has other priorities and i respect that, but the community also needs this tool and the help of the people who are willing to help to continue down this path and keep this tool alive. 

What im trying to say is @Ylianst needs help and there is a certain need for change in a good way and i think this would helpful.

So @Ylianst please consider this!

**Comment by compumatter** [comment]
We are open to the renaming of the project to remotematter and forking the project at https://github.com/compumatter/RemoteMatter

We use the software regularly.  We are a bricks and mortar computer store (compumatter) that have had a physical location in Washington State (Anacortes) for 18 years.  I've personally been in programming / development for decades though NodeJS is not a specialty though I have great respect for it.   I've been in wide variety of technology for a very long time.

We have been developing a small business server project for the past few years and anticipating a launch of this product which will include MeshCentral (rebranded RemoteMatter)

![image](https://github.com/Ylianst/MeshCentral/assets/121611/39fc2922-5c2a-46df-99b7-e79ec34675cd)

ServerMatter Channel:  https://www.youtube.com/@servermatter4465/videos
CompuMatter Channel: https://www.youtube.com/@compumatterllc/videos

Honestly I have a plate full all the time and did not want to take this step right now.  That said, this software cannot die on the vine and I am willing to do my part to see that it does not.  It is the best in the industry and will live on our server project.  I can see it sliding so I'd better "pipe up" @silversword411 says.

I would expect to be incorporating some kind of sponsoring aspect to this project at some point both to pay developers and perhaps in the future to have an identical for profit motive to keep it viable and ongoing as Nextcloud does.  No difference, same product, but allows it to grow.

If @Ylianst or @krayon007 would like to reach out to me I am happy to interact with them further on it.  If a for profit version evolves, they would naturally be a part of that.

I'll be keeping an ear to the ground....
